src/services/item/plugins/html/h5p/service.ts

```diff
@@ -0,0 +1,89 @@
+import path from 'path';
+import { v4 } from 'uuid';
+
+import { H5PItemExtra, H5PItemType } from '@graasp/sdk';
+
+import { Repositories } from '../../../../../utils/repositories';
+import FileService from '../../../../file/service';
+import { Actor, Member } from '../../../../member/entities/member';
+import { HtmlService } from '../service';
+import { HtmlValidator } from '../validator';
+import { H5P_FILE_DOT_EXTENSION } from './constants';
+import { H5P } from './validation/h5p';
+
+/**
+ * Implementation for the H5P service
+ */
+export class H5PService extends HtmlService {
+  constructor(
+    fileService: FileService,
+    pathPrefix: string,
+    mimetype: string,
+    validator: HtmlValidator,
+  ) {
+    super(fileService, pathPrefix, mimetype, 'h5p', validator);
+  }
+
+  /**
+   * Helper to create H5P extra
+   */
+  buildH5PExtra(contentId: string, filename: string): H5PItemExtra {
+    return {
+      h5p: {
+        contentId,
+        h5pFilePath: this.buildPackagePath(contentId, filename),
+        contentFilePath: this.buildContentPath(contentId),
+      },
+    };
+  }
+
+  /**
+   * Helper to build the local or remote path of the .h5p file
+   */
+  buildH5PPath = (rootPath: string, filename: string) =>
+    path.join(rootPath, `${filename}.${H5P.H5P_FILE_EXTENSION}`);
+
+  /**
+   * Download the H5P file referenced by a given Item
+   */
+  downloadH5P(item: H5PItemType, member: Actor, destinationPath: string) {
+    const h5pPath = item.extra.h5p.h5pFilePath;
+    return this.fileService.download(member, {
+      id: item.id,
+      path: path.join(this.pathPrefix, h5pPath),
+      mimetype: this.mimetype,
+      fileStorage: destinationPath,
+    });
+  }
```

Comment:
Should this be a general feature of the HTML service? i.e. the service coud impose that you can always store and download the original asset / archive ?

Reply:
I totally agree, but because of the type and path it's a bit specific. I've opted for a `super.download` and each specific service still have to implement their own download. Let me know what you think. 
Otherwise having an abstract function in the abstract class could also work.